File: weather_script.js
const apiKey = '888ae371c0f61f55971d752c22733538'; // Replace with your OpenWeatherMap API key

        document.getElementById('getWeather').addEventListener('click', async () => {
            const city = document.getElementById('city').value;
            if (!city) {
                alert('Please enter a city name.');
                return;
            }

            const url = `https://api.openweathermap.org/data/2.5/weather?q=${city}&appid=${apiKey}&units=metric`;

            try {
                const response = await fetch(url);
                if (!response.ok) {
                    throw new Error('City not found', response.status);
                }

                const data = await response.json();
                const weather = `
                    <h2>Weather in ${data.name}</h2>
                    <p>Temperature: ${data.main.temp}°C</p>
                    <p>Weather: ${data.weather[0].description}</p>
                    <p>Humidity: ${data.main.humidity}%</p>
                    <p>Wind Speed: ${data.wind.speed} m/s</p>
                `;
                document.getElementById('weatherResult').innerHTML = weather;
            } catch (error) {
                document.getElementById('weatherResult').innerHTML = `<p>Error: ${error.message}</p>`;
            }
        });

File: weather_script.py
import requests

def fetch_weather(city, api_key):
    base_url = "http://api.openweathermap.org/data/2.5/weather"
    params = {
        "q": city,
        "appid": api_key,
        "units": "metric"  # Use "imperial" for Fahrenheit
    }
    try:
        response = requests.get(base_url, params=params)
        response.raise_for_status()  # Raise an HTTPError for bad responses (4xx and 5xx)
        data = response.json()
        return {
            "city": data["name"],
            "temperature": data["main"]["temp"],
            "description": data["weather"][0]["description"]
        }
    except requests.exceptions.RequestException as e:
        return f"Error fetching weather data: {e}"

def main():
    print("Weather Fetcher")
    city = input("Enter the city name: ")
    api_key = input("Enter your OpenWeatherMap API key: ")

    weather = fetch_weather(city, api_key)
    if isinstance(weather, dict):
        print(f"Weather in {weather['city']}:")
        print(f"Temperature: {weather['temperature']}°C")
        print(f"Description: {weather['description']}")
    else:
        print(weather)

if __name__ == "__main__":
    main()